### PR TITLE
crimson/os: do not fail if fsid file exists when mkfs

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
cc @liu-chunmei 

this change is part of #26697

File: src/crimson/os/cyan_store.h [hunk 1]
```diff
@@ -37,6 +37,11 @@ class CyanStore {
 				   uint64_t offset,
 				   size_t len,
 				   uint32_t op_flags = 0);
+  seastar::future<ceph::bufferptr> get_attr(CollectionRef c,
+					    const ghobject_t& oid,
+					    string_view name);
```
Comment (rzarzynski) [inline src/crimson/os/cyan_store.h]: May we have the full namespace for  the `string_view`?

File: src/crimson/osd/osd.cc [hunk 2]
```diff
@@ -78,13 +78,11 @@ seastar::future<> OSD::mkfs(uuid_d cluster_fsid)
...
-  uuid_d osd_fsid;
-  osd_fsid.generate_random();
-  return store->mkfs(osd_fsid).then([this] {
+  return store->mkfs().then([this] {
     return store->mount();
-  }).then([cluster_fsid, osd_fsid, this] {
+  }).then([cluster_fsid, this] {
     superblock.cluster_fsid = cluster_fsid;
-    superblock.osd_fsid = osd_fsid;
+    superblock.osd_fsid = store->get_fsid();
```
Comment (rzarzynski) [inline src/crimson/osd/osd.cc]: This commit looks fine.

File: src/crimson/os/cyan_store.h (hunk 1 above)
Comment (tchaikov) [reply src/crimson/os/cyan_store.h]: fixed and repushed.

**Pull request merged**